src/main/java/frc/robot/subsystems/arm/ArmSystemModel.java

```diff
@@ -0,0 +1,161 @@
+package frc.robot.subsystems.arm;
+
+import frc.robot.utils.motors.FalconConstants;
+import org.ejml.data.MatrixType;
+import org.ejml.simple.SimpleMatrix;
+
+/**
+ * This class is used to calculate the feedforward for the arm.
+ * Notice that this class does not calculate velocities or accelerations,
+ * and relies on the user to do so (in the subsystem class).
+ * <p>
+ * Documentation for the math used in this class can be found here:
+ * https://www.chiefdelphi.com/t/whitepaper-two-jointed-arm-dynamics/423060?u=dan
+ */
+public class ArmSystemModel {
+    private final double m1;
+    private final double m2;
+    private final double l1;
+    private final double r1;
+    private final double r2;
+    private final double I1;
+    private final double I2;
+    private final double g1;
+    private final double g2;
+
+    private final SimpleMatrix M_MATRIX = new SimpleMatrix(2, 2, MatrixType.DDRM);
+    private final SimpleMatrix C_MATRIX = new SimpleMatrix(2, 2, MatrixType.DDRM);
+    private final SimpleMatrix Tg_VECTOR = new SimpleMatrix(2, 1, MatrixType.DDRM);
+    private final SimpleMatrix Kb_MATRIX = new SimpleMatrix(2, 2, MatrixType.DDRM);
+    private final SimpleMatrix B_MATRIX = new SimpleMatrix(2, 2, MatrixType.DDRM);
+
+    /**
+     * The constructor for the system model.
+     *
+     * @param systemConstants the system constants of the arm.
+     */
+    public ArmSystemModel(SystemConstants systemConstants) {
+        var shoulderJointConstants = systemConstants.getShoulderJointConstants();
+        var elbowJointConstants = systemConstants.getElbowJointConstants();
```

Comment:
This is the only place the `SystemConstants` class is used, and you just unpack it here (so you don't really use the joined (😉) class.
I'd delete it and use the `JointConstants` objects directly.